src/latexify/codegen/function_codegen.py

```diff
@@ -162,21 +162,47 @@ def visit_Match(self, node: ast.Match) -> str:
             if i < len(node.cases) - 1:
                 body_latex = self.visit(case.body[0])
                 cond_latex = self.visit(case.pattern)
-                case_latexes.append(
-                    body_latex + r", & \mathrm{if} \ " + subject_latex + cond_latex
-                )
+                if case.guard is not None:
+                    cond_latex = self._expression_codegen.visit(case.guard)
+
+                case_latexes.append(body_latex + r", & \mathrm{if} \ " + cond_latex)
             else:
                 case_latexes.append(
-                    self.visit(node.cases[-1].body[0]) + r", & \mathrm{otherwise}"
+                    self.visit(case.body[0]) + r", & \mathrm{otherwise}"
                 )
 
-        return (
-            r"\left\{ \begin{array}{ll} "
+        latex = (
+            r"\left\{ \begin{array}{ll}"
             + r" \\ ".join(case_latexes)
             + r" \end{array} \right."
         )
 
+        latex_final = latex.replace("subject_name", subject_latex)
+        return latex_final
+
     def visit_MatchValue(self, node: ast.MatchValue) -> str:
-        """Visit a MatchValue node"""
+        """Visit a MatchValue node."""
         latex = self._expression_codegen.visit(node.value)
-        return " = " + latex
+        return "subject_name = " + latex
+
+    def visit_MatchAs(self, node: ast.MatchAs) -> str:
+        """
+        Visit a MatchAs node.
+        If MatchAs is a wildcard, return 'otherwise' case, otherwise throw an error.
+        """
+        if node.pattern is None:
+            return ""
+        else:
+            raise exceptions.LatexifySyntaxError(
+                "Nonempty as-patterns are not supported in MatchAs nodes."
+            )
+
+    def visit_MatchOr(self, node: ast.MatchOr) -> str:
+        """Visit a MatchOr node."""
+        case_latexes = []
+        for i, pattern in enumerate(node.patterns):
+            if i == 0:
+                case_latexes.append(self.visit(pattern))
+            else:
+                case_latexes.append(r" \lor " + self.visit(pattern))
+        return "".join(case_latexes)
```

Comment:
Too redundant. Joining all conditions with `\lor` is much efficient.
```suggestion
        return r" \lor ".join(self.visit(p) for p in node.patterns)
```